src/renderer/api-decorator.js

```diff
@@ -560,52 +560,93 @@ limitations under the License.
-                        }
-                    }
-                });
-            }
+        if (plugin.length) {
+            evalPlugins(identity, plugin);
         }
 
-        asyncApiCall(action, { allDone: true });
+        if (preload.length) {
+            evalPreloadScripts(identity, preload);
+        }
     });
 
-    function logPreload(level, identity, state, url, data) {
-        if (url) {
-            state += ` for ${url}`;
+    /**
+     * Requests plugin contents from the Core and evals them in the current window
+     */
+    function evalPlugins(identity, pluginOption) {
+        const action = 'set-window-plugin-state';
+        let logBase = `[plugin] [${identity.uuid}]-[${identity.name}]: `;
+        let plugins;
+
+        try {
+            plugins = syncApiCall('get-selected-preload-scripts', pluginOption);
+        } catch (error) {
+            return syncApiCall('write-to-log', { level: 'error', message: logBase + error });
         }
-        if (data) {
-            state += ` with ${JSON.stringify(data)}`;
+
+        plugins.forEach((plugin) => {
+            const { name, version, content } = plugin;
+
+            if (content !== null) {
+                // TODO: handle empty script for bad urls
+
+                try {
+                    window.eval(content); /* jshint ignore:line */
+                    asyncApiCall(action, { name, version, state: 'succeeded' });
+                    syncApiCall('write-to-log', {
+                        level: 'info',
+                        message: logBase + `eval succeeded for ${name} ${version}`
+                    });
+                } catch (err) {
+                    asyncApiCall(action, { name, version, state: 'failed' });
+                    syncApiCall('write-to-log', {
+                        level: 'info',
+                        message: logBase + `eval failed for ${name} ${version}`
+                    });
+                }
+            }
+        });
+
+        asyncApiCall(action, { allDone: true });
+    }
+
+    /**
+     * Requests preload scripts contents from the Core and evals them in the current window
+     */
+    function evalPreloadScripts(identity, preloadOption) {
+        const action = 'set-window-preload-state';
+        let logBase = `[preload] [${identity.uuid}]-[${identity.name}]: `;
+        let preloadScripts;
+
+        try {
+            preloadScripts = syncApiCall('get-selected-preload-scripts', preloadOption);
+        } catch (error) {
+            return syncApiCall('write-to-log', { level: 'error', message: logBase + error });
         }
-        const message = `[PRELOAD] [${identity.uuid}]-[${identity.name}] ${state}`;
-        syncApiCall('write-to-log', { level, message });
+
+        preloadScripts.forEach((preloadScript) => {
+            const { url, content } = preloadScript;
+
+            if (content !== null) {
+                // TODO: handle empty script for bad urls
```

Comment:
What happens currently, just that failed eval below? (just asking)

Reply:
right now this is not working (with and without change)... like many other things... will be fixing